src/include/migraphx/op/topk.hpp

```diff
@@ -0,0 +1,110 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_GATHER_HPP
+#define MIGRAPHX_GUARD_OPERATORS_GATHER_HPP
+
+#include <algorithm>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/heap.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/value.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct topk
+{
+    int64_t k    = 1;
+    int64_t axis = 0;
+    bool largest = true;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.k, "k"), f(self.axis, "axis"), f(self.largest, "largest"));
```

Comment:
The reason is: we call the heap function in the stl lib for the ref implementation, but we cannot do that for the GPU implementation, so we have to implement the heap function ourselves. Because we do not know the specific algorithm used in the stl lib, so if we do not sort the output, there is no guarantee that the ref and gpu implementation have the same outputs, and our GPU tests will fail. To avoid that, we always sort the outputs.